docs: update java serialize guide (#173)
apache/fury-site: docs/guide/java_serialization_guide.md

### Implement a customized serializer

> Implementing serialized classes, JDK serialization is very performance and space inefficient. For example, if you don't want `Foo#writeObject` below to be called, you can implement `FooSerializer` below the type:

Reviewer: ```suggestion
In some cases, you may want to implement a serializer for your type, especially for classes customized serialization by JDK serialization. JDK serialization is very inefficient in performance and space. You can write your own Serializer and configure Fury to used it for serialization instead.

For example, if you don't want belowing `Foo#writeObject` to be called, you can implement and register `FooSerializer`:
```

### Security & Class Registration

> `FuryBuilder#requireClassRegistration` can be used to disable class registration, this will allow to deserialize objects unknown types, more flexible but ****if the class contains malicious code, a security breach can occur**.**.

Reviewer: ```suggestion
`FuryBuilder#requireClassRegistration` can be used to disable class registration, this will allow to deserialize objects unknown types, more flexible but ****if the class contains malicious code, a security vulnerability can occur**.**.l
```

**Do not disable class registration checking unless you can ensure the security of your runtime environment and external interactions**.

> Malicious code in `init/equals/hashCode` can be executed when deserializing unknown/untrusted types when this option disabled.

Reviewer: ```suggestion
Malicious code in `init/equals/hashCode` can be executed when deserializing unknown/untrusted types when this option is disabled.
```

If metadata sharing is not enabled, the new class data will be skipped and an `NonexistentSkipClass` stub object will be
returned.

> ## Serialized Library Migration

Reviewer: ```suggestion
## Serialization Library Migration
```

### JDK migration

> If you use JDK serialization before, and you can't upgrade your client and server at the same time, which is common for online application. Fury provided an util method `org.apache.fury.serializer.JavaSerializer.serializedByJDK` to check whether the binary are generated by jdk serialization, you use following pattern to make exiting serialization protocol-aware, then upgrade serialization to fury in an async rolling-up way:

Reviewer: ```suggestion
If you use JDK serialization before, and you can't upgrade your client and server at the same time, which is common for online application. Fury provided an util method `org.apache.fury.serializer.JavaSerializer.serializedByJDK` to check whether the binary are generated by jdk serialization, you use following pattern to make existing serialization protocol-aware, then upgrade serialization to fury in a rolling-up way:
```

### Upgrade fury

> Currently binary compatibility is ensured for minor versions only. For example, if you are using fury`v0.2.0`, binary compatibility will be provided if you upgrade to fury `v0.2.1`. But if upgrade to fury `v0.4.1`, no binary compatibility are ensured.

Reviewer: ```suggestion
Currently binary compatibility is ensured for minor versions only. For example, if you are using fury`v0.2.0`, binary compatibility will be provided if you upgrade to fury `v0.2.1`. But if you upgrade fury to `v0.4.1`, no binary compatibility are ensured.
```

  fury.deserialize(buffer);
```

`getFury` is the way to load the corresponding version of Fury. You can use the maven shade plugin to shade different versions of Fury and relocate them under different packages, so that you can load different versions of Fury under different paths.

If you upgrade fury by minor version, or you won't have data serialized by older fury, you can upgrade fury directly, no need to `versioning` the data.

> ## Troubleshooting Common Problems

Reviewer: ```suggestion
## Troubleshooting Common Issues
```
